src/app/modules/form/form.utils.ts

```diff
@@ -76,3 +79,37 @@ export const isEmailModeForm = (
 ): form is IPopulatedEmailForm => {
   return form.responseMode === ResponseMode.Email
 }
+
+/**
+ * Type guard for whether given array is a mongoose DocumentArray
+ * @param array the array to check
+ */
+export const isMongooseDocumentArray = <T extends Document>(
+  array: T[] & { isMongooseDocumentArray?: boolean },
+): array is Types.DocumentArray<T> => {
+  /**
+   * @see {mongoose.Types.DocumentArray.isMongooseDocumentArray}
+   */
+  return !!array.isMongooseDocumentArray
+}
+
+/**
+ * Finds and returns form field in given form by its id
+ * @param formFields the form fields to search from
+ * @param fieldId the id of the field to retrieve
+ * @returns the form field if found, `null` otherwise
+ */
+export const getFormFieldById = (
+  formFields: IFormSchema['form_fields'],
+  fieldId: IFieldSchema['_id'],
+): IFieldSchema | null => {
+  if (!formFields) {
+    return null
+  }
+
+  if (isMongooseDocumentArray(formFields)) {
+    return formFields.id(fieldId)
+  } else {
+    return formFields.find((f) => fieldId === String(f._id)) ?? null
+  }
```

Comment:
since when did you use `else`??? lol

Reply:
i am disgusted with myself; fixed in ea629fd